### DISCUSSION TITLE
px4 STIL + Unreal + QGroundControl

### DISCUSSION THREAD

**Discussion by FalkonEight** [microsoft/AirSim#4135] [Support Q&A]
## Question
Hi, I configured px4 on wsl2 and environment in unreal, everything connects correctly, but I can not connect to QGroundControl please advise how to do it correctly

**Comment by jonyMarino** [comment]
Hi @FightForDobro. That is a PX4/QGroundControl question, but also a question of interest for AirSim users.

**Comment by jonyMarino** [comment]
QGroundControl is not currently supported in WSL. So the question should be how to connect QGroundControl on Windows to a PX4 instance running on WSL:
For that, you need to :
1_ clone [the mavlink router repository](https://github.com/mavlink-router/mavlink-router/) inside your WSL instance and follow the instructions there (clone it, do not download the source from the releases page).
2_ run the following command inside WSL:
mavlink-routerd -e [WSL_IP]:14550 127.0.0.1:14550
you can get the WSL IP doing: cat /etc/resolv.conf | grep nameserver | awk '{print $2}' inside WSL.
